test/snippet/core/algorithm/configuration.cpp

```diff
@@ -1,88 +1,96 @@
+#include <iostream>
+
+//! [configuration_setup]
 #include <seqan3/core/algorithm/configuration.hpp>
+#include <seqan3/core/algorithm/pipeable_config_element.hpp>
 
 using namespace seqan3;
 
-struct bar
+enum struct my_id : int
 {
-    float value = 0;
+    bar_id,
+    foo_id
 };
 
-struct foo
+struct bar : public pipeable_config_element
```

Comment:
Since `bar` has a non-default constructor it should be a class (you also talk about classes in your description). Same goes for foo then.